objax/nn/layers.py

```diff
@@ -166,6 +166,7 @@ def __init__(self,
         super().__init__()
         assert nin % groups == 0, 'nin should be divisible by groups'
         assert nout % groups == 0, 'nout should be divisible by groups'
+        self.nin = nin
```

Comment:
I am not sure we need to add `self.nin` when it can be inferred from `self.w`.

E.g.
```python
@property
def nin(self):
    return self.w.value.shape[-2]
```

What do you think?

Reply:
Is there a reason this is better? A convolution should never hav w.value change shape so I don't think nin should ever go stale. It's more lines of code and uses property magic that I'd avoid by default unless there's a reason you think nin might sometimes be wrong?

Reply:
The only reason is coherency and I am not sure about it. Let's just keep `self.nin = nin` since it's a lot simpler as you pointed out.